dspace-api/src/main/java/org/dspace/curate/Curator.java

```diff
@@ -268,7 +268,22 @@ else if (type == Constants.SITE)
             }
         }
     }
-    
+
+    /**
+     * Performs all configured tasks upon DSpace object
+     * (Community, Collection or Item).
+     *
+     * @param c session context in which curation takes place.
+     * @param dso the single object to be curated.
+     * @throws java.io.IOException passed through.
+     */
+    public void curate(Context c, DSpaceObject dso)
+            throws IOException
+    {
+        curationCtx.set(c);
```

Comment:
Ugh, this has the side-effect of leaving the Curator set to the given Context.  Should we save the old reference and restore it before return?

Or should Curator have a setContext(Context) instead?